sphinx/builders/__init__.py

```diff
@@ -635,7 +640,7 @@ def read_doc(self, docname: str, *, _cache: bool = True) -> None:
         self.env.all_docs[docname] = time.time_ns() // 1_000
 
         # cleanup
-        self.env.temp_data.clear()
```

Comment:
Maybe we should keep this one if people add stuff in their own temp_data and expect it to be cleaned up.